google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -3589,4 +3589,84 @@ HmacKeyMetadata updateHmacKeyState(
    * @throws StorageException upon failure
    */
   ServiceAccount getServiceAccount(String projectId);
+
+  /**
+   * Adds a new notification to the bucket.
+   *
+   * <p>Example of adding a notification:
+   *
+   * <pre>{@code
+   * String bucketName = "my-unique-bucket";
+   * String topic = "projects/myProject/topics/myTopic"
+   * Notification bucketNotification = Notification.newBuilder(topic)
+   *  .setCustomAttributes(ImmutableMap.of("label1", "value1"))
+   *  .setEventTypes(Notification.EventType.OBJECT_FINALIZE)
+   *  .setPayloadFormat(Notification.PayloadFormat.JSON_API_V1)
+   *  .build();
```

Comment:
The issue with the current flow is that Notification instances are initialized using a topic but not the id. 

Additionally, I can't tell when I have an object that's returned by the Storage API or the library. For example BlobInfo is user generated while Blob is returned by the Storage API. The same pattern is used for BucketInfo and Bucket. 

I think the path forward would be to do something similar. Pending input @JesseLovelace.

Reply:
okay so do you like me to add the similar kind of pattern here correct?  for example `NotificationInfo `is user generated while `Notification `is returned by the Storage API.